lib/listen/record.rb

```diff
@@ -75,31 +63,36 @@ def build
 
     private
 
+    # TODO: refactor/refactor out
+    def add_dir(dir, rel_path, children)
+      rel_path = '.' if [nil, '', '.'].include? rel_path
+      dirname, basename = Pathname(rel_path).split.map(&:to_s)
+      basename = '.' if [nil, '', '.'].include? basename
+      if [nil, '', '.'].include?(dirname) && basename == '.'
+        @paths[dir.to_s]['.'] = Hash[children.map { |child| [child, {} ] } ]
+      else
+        @paths[dir.to_s][rel_path] ||= {}
```

Comment:
@driskell - I'm just on the fence about it, because so many other things are wrong here. It simplifies the code, but complicates the internal structure - which can get in the way if `Record` is to handle directories. The `Record` tests need rework anyway.

Just assume that '.' is fine for now - and I take back what I said about it.